tests/initcoder_test.go

```diff
@@ -0,0 +1,184 @@
+package tests
+
+import (
+	"context"
+	"math/big"
+	"testing"
+
+	"github.com/Fantom-foundation/go-opera/tests/contracts/contractcreator"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/require"
+)
+
+// this constant comes from  https://eips.ethereum.org/EIPS/eip-3860#parameters
+const MAX_INIT_CODE_SIZE uint64 = 49152
+
+const sufficientGas = uint64(100_000)
+
+func TestInitCodeSizeLimitAndMetered(t *testing.T) {
+	requireBase := require.New(t)
+
+	net, err := StartIntegrationTestNet(t.TempDir())
+	requireBase.NoError(err)
+	defer net.Stop()
+
+	contract, receipt, err := DeployContract(net, contractcreator.DeployContractcreator)
+	requireBase.NoError(err)
+	requireBase.Equal(types.ReceiptStatusSuccessful, receipt.Status, "failed to deploy contract")
+
+	// deploy measureGasAndAssign to get cost of deploying a contract without create.
+	receipt = createContractSuccessfully(t, net, contract.GetOverheadCost, 0, 0)
+
+	// -- using CREATE instruction
+	const wordCostCreate uint64 = 2
+	var gasForCreate uint64 = 32035 + receipt.GasUsed //+ assignCost
```

Comment:
nit: this comment is not clear, is this already added, or is it a comment to the formula?